server/src/main/java/com/defold/extender/services/CocoaPodsService.java

```diff
@@ -720,14 +744,130 @@ private void parseMultiPlatformConfig(PodSpec spec, JSONObject config) {
+        LOGGER.info("createModuleMap() " + pod.moduleName +  " from header patterns: " + headerPatterns + " " + moduleMapFile);
+
+        List<String> headers = new ArrayList<>();
+        List<String> umbrellaHeaders = new ArrayList<>();
+        List<String> umbrellaDirectories = new ArrayList<>();
+
+        // swift source file header
+        // generated by the swift compiler if -emit-objc-header flag is set
+        if (!pod.swiftSourceFiles.isEmpty()) {
+            File swiftModuleHeader = new File(pod.generatedDir, pod.moduleName + "-Swift.h");
+            pod.swiftModuleHeader = swiftModuleHeader;
+            headers.add(swiftModuleHeader.getAbsolutePath());
+        }
+
+        // umbrella headers
+        for (String headerPattern : headerPatterns) {
+            String absoluteHeaderPatternPath = pod.dir.getAbsolutePath() + File.separator + headerPattern;
+            List<File> headerFiles = listFilesGlob(pod.dir, absoluteHeaderPatternPath);
+            for (File headerFile : headerFiles) {
+                if (headerFile.getName().equals(pod.moduleName + ".h")) {
+                    umbrellaHeaders.add(headerFile.getAbsolutePath());
+                }
+            }
+        }
+
+        // umbrella directory
+        if (umbrellaHeaders.isEmpty()) {
+            for (String headerPattern : headerPatterns) {
+                headerPattern = headerPattern.replace("/*.h", "");
+                headerPattern = headerPattern.replace("/**", "");
+                if (headerPattern.lastIndexOf("/") != -1) {
+                    headerPattern = headerPattern.substring(0, headerPattern.lastIndexOf("/"));
+                }
+                String headerPath = pod.dir.getAbsolutePath() + File.separator + headerPattern;
+                umbrellaDirectories.add(headerPath);
+            }
+        }
+
+        // module name, including parent spec module name
+        String fullModuleName = ((pod.parentSpec != null) ? (pod.parentSpec.moduleName + "_") : "") + pod.moduleName;
+
+        // generate final modulemap
+        HashMap<String, Object> envContext = new HashMap<>();
+        envContext.put("FRAMEWORKOPT", pod.containsFramework ? "framework" : "");
+        envContext.put("MODULE_ID", fullModuleName);
+        envContext.put("HEADERS", headers);
+        envContext.put("UMBRELLA_HEADERS", umbrellaHeaders);
+        envContext.put("UMBRELLA_DIRECTORIES", umbrellaDirectories);
+        envContext.put("SUBMODULE", pod.containsFramework ? "module * { export * }" : "");
+        
+        String moduleMap = templateExecutor.execute(modulemapTemplateContents, envContext);
+        try {
+            Files.writeString(moduleMapFile.toPath(), moduleMap);
+        }
+        catch (IOException e) {
+            throw new ExtenderException(e, "Unable to create modulemap for " + pod.moduleName);
+        }
+    }
+
+    private File createModuleMapFile(PodSpec pod, String platform) {
```

Comment:
The platform name isn't strictly really necessary for the filename?
(Each build will get its own build folder)